cmd/server.go

```diff
@@ -195,15 +195,23 @@ func NewServerCommand(stopCh <-chan struct{}) *cobra.Command {
 					}
 					if err != nil {
 						if etcdErr, ok := err.(*errors.EtcdError); ok == true {
-							logger.Errorf("Failed to take first delta snapshot: snapshotter failed with etcd error: %v", etcdErr)
+							logger.Warnf("Failed to take first delta snapshot: snapshotter failed with etcd error: %v", etcdErr)
 						} else {
-							logger.Errorf("Failed to take first delta snapshot: snapshotter failed with error: %v", err)
+							logger.Warnf("Failed to take first delta snapshot: snapshotter failed with error: %v", err)
+						}
+						startWithFullSnapshot = false
+						if err := ssr.TakeFullSnapshotAndResetTimer(); err != nil {
```

Comment:
Why not `continue` only on delta-snapshot success and fall-through here?

Reply:
That would mean we wait for the snapshotter take the first snapshot (which will be a full snapshot) and set `handler.Status = http.StatusOK` before that, without actually having taken a snapshot before. We want to take atleast one snapshot (delta or full) before setting the readiness probe to true, so that we know that sidecar is able to take snapshots and is actually ready.

Reply:
@shreyas-s-rao Agreed. But I guess Amshu's concern if more about complex if-else ladder here. If not atleaset i find it little complex. I think we can simplify it by reversing the if checks and have ssr.TakeFullSnapshotAndResetTimer() call at single place with minimum if-else conditions. e.g.  check at https://github.com/gardener/etcd-backup-restore/pull/165/files#diff-1427b112aa72f64adcf839007e954d56R180 can be reversed to evaluate else case first. Also, later on https://github.com/gardener/etcd-backup-restore/pull/165/files#diff-1427b112aa72f64adcf839007e954d56R196 this kind of checks can be changed to `err == nil`.  Please give it try.

Reply:
Yes, @amshuman-kr had the same suggestion in mind. I'll make this change. 👍🏼